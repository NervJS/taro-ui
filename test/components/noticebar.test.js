import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtNoticebar from '../../.temp/components/noticebar/index'

describe('AtNoticebar Snap', () => {
  it('render AtNoticebar -- props show', () => {
    const component = renderToString(<AtNoticebar close>这是内容</AtNoticebar>)
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props single', () => {
    const component = renderToString(<AtNoticebar single>这是内容</AtNoticebar>)
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props speed', () => {
    const component = renderToString(
      <AtNoticebar speed={200}>这是内容</AtNoticebar>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props moreText & showMore', () => {
    const component = renderToString(
      <AtNoticebar showMore moreText='查看更多'>
        这是内容
      </AtNoticebar>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props icon', () => {
    const component = renderToString(
      <AtNoticebar icon='volume-plus'>这是内容</AtNoticebar>
    )
    expect(component).toMatchSnapshot()
  })
})

describe('AtNoticebar Event', () => {
  it('AtNoticebar onClose', () => {
    const onClose = jest.fn()
    const component = renderIntoDocument(
      <AtNoticebar icon='volume-plus' onClose={onClose} close>
        这是内容
      </AtNoticebar>
    )
    const dom = findDOMNode(component, 'at-noticebar').querySelector(
      '.at-noticebar__close'
    )
    Simulate.click(dom)
    process.nextTick(() => {
      expect(onClose).toBeCalled()
    })
  })

  it('AtNoticebar onGotoMore', () => {
    const onGotoMore = jest.fn()
    const component = renderIntoDocument(
      <AtNoticebar
        icon='volume-plus'
        onGotoMore={onGotoMore}
        single
        showMore
        moreText='更多内容'
      >
        这是内容
      </AtNoticebar>
    )
    const dom = findDOMNode(component, 'at-noticebar').querySelector(
      '.at-noticebar__more'
    )
    Simulate.click(dom)
    process.nextTick(() => {
      expect(onGotoMore).toBeCalled()
    })
  })
})
