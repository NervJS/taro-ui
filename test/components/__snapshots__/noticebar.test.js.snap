// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtNoticebar Snap render AtNoticebar -- props icon 1`] = `"<div class=\\"at-noticebar\\">undefined<div class=\\"at-noticebar__content\\"><div class=\\"at-noticebar__content-icon\\"><span class=\\"taro-text at-icon at-icon-volume-plus at-icon at-icon-volume-plus\\"></span></div><div class=\\"at-noticebar__content-text\\"><div class=\\"at-noticebar__content-inner\\">这是内容</div></div></div>undefined</div>"`;

exports[`AtNoticebar Snap render AtNoticebar -- props moreText & showMore 1`] = `"<div class=\\"at-noticebar\\">undefined<div class=\\"at-noticebar__content\\"><div class=\\"at-noticebar__content-text\\"><div class=\\"at-noticebar__content-inner\\">这是内容</div></div></div>undefined</div>"`;

exports[`AtNoticebar Snap render AtNoticebar -- props show 1`] = `"<div class=\\"at-noticebar\\"><div class=\\"at-noticebar__close\\"><span class=\\"taro-text at-icon at-icon-close\\"></span></div><div class=\\"at-noticebar__content\\"><div class=\\"at-noticebar__content-text\\"><div class=\\"at-noticebar__content-inner\\">这是内容</div></div></div>undefined</div>"`;

exports[`AtNoticebar Snap render AtNoticebar -- props single 1`] = `"<div class=\\"at-noticebar at-noticebar--single\\">undefined<div class=\\"at-noticebar__content\\"><div class=\\"at-noticebar__content-text\\"><div class=\\"at-noticebar__content-inner\\">这是内容</div></div></div>undefined</div>"`;

exports[`AtNoticebar Snap render AtNoticebar -- props speed 1`] = `"<div class=\\"at-noticebar\\">undefined<div class=\\"at-noticebar__content\\"><div class=\\"at-noticebar__content-text\\"><div class=\\"at-noticebar__content-inner\\">这是内容</div></div></div>undefined</div>"`;
